clang/test/InstallAPI/diagnostics-dsym.test

```diff
@@ -0,0 +1,40 @@
+; RUN: rm -rf %t
+; RUN: split-file %s %t
+
+// Build a simple dylib with debug info.
+; RUN: %clang --target=arm64-apple-macos13 -g -dynamiclib %t/foo.c \
```

Comment:
@JDevlieghere Is there a better/known practice for generating full dylibs with debug info for tests? Hopefully, PR CI will let me know, but I'm a little worried about different linker semantics. 
e.g. Xcode's ld requires all userspace dylibs link against libSystem.

Reply:
Looks like the linker on the linux bot doesn't support building Darwin anyway
```
/usr/bin/ld: unrecognised emulation mode: llvm
Supported emulations: elf_x86_64 elf32_x86_64 elf_i386 elf_iamcu elf_l1om elf_k1om i386pep i386pe
clang: error: linker command failed with exit code 1 (use -v to see invocation)
```
Perhaps I'll require Darwin env to run the test and then I won't need to introduce a `dsymutil` dependency at all since it can come from xcode.